### ISSUE TITLE
csm_ras.log file has incorrect timestamp  for events created by Diagnostic

### ISSUE THREAD

**Issue by sanomiya** [IBM/CAST#102]
Diagnostic creates a ras event using csm_ras_event_create common without passing the timestamp.
When I query the database using csm_ras_event_query, the timestamp is displayed correctly.
But in the csm_ras.log files shows :
```
{"time_stamp":"now","msg_id":"hcdiag.test.fail","location_name":"c699c192","raw_data":"","kvcsv":"test=chk-gpfs-mount,node=c699c192,sn=785600A,rc=1,file=\/shared\/logs\/node_health\/180606114724663493\/chk-gpfs-mount\/c699c192-2018-06-06-11_47_27.output","test":"chk-gpfs-mount","node":"c699c192","sn":"785600A","rc":"1","file":"\/shared\/logs\/node_health\/180606114724663493\/chk-gpfs-mount\/c699c192-2018-06-06-11_47_27.output","ctxid":"418","severity":"WARNING","message":"chk-gpfs-mount FAIL on node c699c192 serial number: 785600A (details in \/shared\/logs\/node_health\/180606114724663493\/chk-gpfs-mount\/c699c192-2018-06-06-11_47_27.output).","control_action":"NONE","description":"NONE","enabled":"1","set_state":"","visible_to_users":"1","threshold_count":"1","threshold_period":"5","count":"1"}
```

**Comment by besawn** [comment]
This is due to the csm_ras_event_create front end assigning a string of "now()" to the timestamp field when the caller does not supply a timestamp, and the csm_ras_event_create back end passing the params from the front end through to csm_ras.log without any DB interaction that would cause the "now()" string to be expanded.

The csm_ras_event_create backend will generate a timestamp rather than using "now()" if no timestamp is passed from the front end, but that behavior will never be triggered if the front end sets the string to "now()" rather than leaving it null. 

I think the best solution is probably to modify the csm_ras_event_create front end to generate an actual timestamp if the caller does not pass one instead of setting the string to"now()", but I would like to discuss the details with @NickyDaB to make sure he agrees with this approach.

**Comment by NickyDaB** [comment]
First thing that pops into my mind is what if the clock on the compute node is different than the clock in the database?

**Comment by NickyDaB** [comment]
It's been almost a month. Does anyone want to reply to my question? @sanomiya @mew2057

**Comment by besawn** [comment]
@NickyDaB csm_ras_event_create is already susceptible to the issue that the time_stamp field as supplied by the caller of the API is not using any common clock. My current thinking on this is that time_stamp as supplied by the caller of the API should correlate with the local time and logging of the process calling the API and master_time_stamp gives us information about when various events were processed relative to each other always using the same time source.

I think there are two options here:
1.) The front end should generate a timestamp if the caller does not supply one, which will match the local time of the node from which the event was created.
2.) The front end should just ignore any null timestamps and pass them through to the backend. In this case, the backend will generate a timestamp instead, which will be relative to when the event was processed by the backend.

Either option is better than passing "now()" through.

**Comment by NickyDaB** [comment]
1 sounds like a better option

**Comment by NickyDaB** [comment]
@sanomiya the feature is now in the development branch and will be apart of the next CSM release.

**Comment by NickyDaB** [comment]
@sanomiya Are you satisfied with the new functions and features? Can we close this issue?

**Comment by sanomiya** [comment]
Fixed on 1.2.0.
```
{"time_stamp":"2018-07-31 13:19:40.183352","msg_id":"hcdiag.test.fail","location_name":"launchgr","raw_data":"","kvcsv":"test=chk-gpfs-mount,node=launchgr,sn=10049AA,rc=1,file=\/tmp\/180731181939453471\/chk-gpfs-mount\/launchgr-2018-07-31-18_19_39.output","test":"chk-gpfs-mount","node":"launchgr","sn":"10049AA","rc":"1","file":"\/tmp\/180731181939453471\/chk-gpfs-mount\/launchgr-2018-07-31-18_19_39.output","ctxid":"16577","severity":"WARNING","message":"chk-gpfs-mount FAIL on node launchgr serial number: 10049AA (details in \/tmp\/180731181939453471\/chk-gpfs-mount\/launchgr-2018-07-31-18_19_39.output).","control_action":"NONE","description":"NONE","enabled":"1","set_state":"","visible_to_users":"1","threshold_count":"1","threshold_period":"5","count":"1"}
{"time_stamp":"2018-07-31 13:19:40.298506","msg_id":"hcdiag.test.fail","location_name":"launchgr","raw_data":"","kvcsv":"test=chk-nfs-mount,node=launchgr,sn=10049AA,rc=2,file=\/tmp\/180731181939453471\/chk-nfs-mount\/launchgr-2018-07-31-18_19_40.output","test":"chk-nfs-mount","node":"launchgr","sn":"10049AA","rc":"2","file":"\/tmp\/180731181939453471\/chk-nfs-mount\/launchgr-2018-07-31-18_19_40.output","ctxid":"16578","severity":"WARNING","message":"chk-nfs-mount FAIL on node launchgr serial number: 10049AA (details in \/tmp\/180731181939453471\/chk-nfs-mount\/launchgr-2018-07-31-18_19_40.output).","control_action":"NONE","description":"NONE","enabled":"1","set_state":"","visible_to_users":"1","threshold_count":"1","threshold_period":"5","count":"1"}
```

Closing this issue.
